skrf/network.py

```diff
@@ -4523,6 +4588,8 @@ def s2z(s, z0=50, s_def=S_DEF_DEFAULT):
         # Power-waves. Eq.(19) from [3]
         # Creating diagonal matrices of shape (nports,nports) for each nfreqs
         F, G = npy.zeros_like(s), npy.zeros_like(s)
+        F = F.astype(dtype=npy.complex)
+        G = G.astype(dtype=npy.complex)
```

Comment:
this generates a copy of F, G. Did you run into troubles with the initial type? If yes, to avoid an additional copy, maybe add the dtype= parameter in the zeros_like ?

Reply:
I did run into trouble. I'm not sure why I didn't just put these type definitions in the zeros_like to begin with. It's fixed now.